inc/saiswitch.h

```diff
@@ -60,6 +60,12 @@ typedef enum _sai_switch_oper_status_t
     /** Switch encountered a fatal error */
     SAI_SWITCH_OPER_STATUS_FAILED,
 
+    /** VOQ Switch in critical state with available fabric ports below configured configured threshold */
+    SAI_SWITCH_OPER_STATUS_VOQ_FABRIC_PORT_THRESHOLD_LOW,
+
+    /** VOQ Switch in recovery state with available fabric ports above configured configured threshold. Once the recovery is complete switch transitions to UP state */
+    SAI_SWITCH_OPER_STATUS_VOQ_FABRIC_PORT_THRESHOLD_HIGH,
+
 } sai_switch_oper_status_t;
```

Comment:
In the current form, the callback is defined to be triggered  for numUpFabricPorts < THRESH_LOW or numUpFabricPorts > THRESH_HIGH, and not when numUpFabricPorts <= THRESH_LOW or numUpFabricPorts >= THRESH_HIGH.

Thus, there should not be a risk that both the  _LOW and _HIGH conditions are met simultaneously and SAI adapter needs to set both THRESHOLD_LOW and THRESHOLD_HIGH in the same callback: that won't be possible as sai_switch_oper_status_t is an enum not a bitmask. 

Thus, might be good idea for SAI_SWITCH_ATTR_FABRIC_PORT_THRESHOLD_LOW description to be more explicit about semantics being "less then" instead of "less than equal to".

Reply:
Thinking more about it, given the PR defines the new attribute to trigger callback when "above" or "below" threshold, we are good here, pls treat this comment as resolved.